spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -352,4 +359,77 @@ public void testWindowStatements() {
         .isEqualTo(3677); // COUNT(word) OVER (PARTITION BY corpus ORDER BY corpus_date)
     assertThat(row.get(10)).isEqualTo(3677); // COUNT(word) OVER count_window
   }
+
+  /** Method to create a test table of schema NumStruct, in test dataset */
+  protected void writeTestDatasetToBigQuery() {
+    Dataset<Row> df =
+        spark
+            .createDataset(
+                Arrays.asList(
+                    new NumStruct(
+                        1L,
+                        2L,
+                        3L,
+                        ImmutableList.of(new StringStruct("1:str3", "2:str1", "3:str2"))),
+                    new NumStruct(
+                        2L,
+                        3L,
+                        4L,
+                        ImmutableList.of(new StringStruct("2:str3", "3:str1", "4:str2")))),
+                Encoders.bean(NumStruct.class))
+            .toDF();
+    df.write()
+        .format("bigquery")
+        .mode(SaveMode.Append)
+        .option("table", testDataset.toString() + "." + testTable)
+        .option("temporaryGcsBucket", TestConstants.TEMPORARY_GCS_BUCKET)
+        .option("writeMethod", WriteMethod.INDIRECT.toString())
+        .save();
+  }
+
+  @Test
+  public void testAggregateExpressions() {
+    writeTestDatasetToBigQuery();
+    Dataset<Row> df =
+        spark
+            .read()
+            .format("bigquery")
+            .option("materializationDataset", testDataset.toString())
+            .load(testDataset.toString() + "." + testTable);
+
+    df.createOrReplaceTempView("numStructDF");
+
+    List<Row> result =
+        spark
+            .sql(
+                "SELECT "
+                    + "AVG(num1) as average, "
+                    + "CORR(num1, num2) as corr, "
+                    + "COVAR_POP(num1, num2) as covar_pop, "
+                    + "COVAR_SAMP(num1, num2) as covar_samp, "
+                    + "COUNT(*) as count, "
+                    + "MAX(num1) as max, "
+                    + "MIN(num1) as min, "
+                    + "SUM(num1) as sum, "
+                    + "STDDEV_POP(num1) as stddev_pop, "
+                    + "ROUND(STDDEV_SAMP(num1),2) as stddev_samp, "
+                    + "VAR_POP(num1) as var_pop, "
+                    + "VAR_SAMP(num1) as var_samp "
+                    + "FROM numStructDF ")
+            .collectAsList();
+
+    Row r1 = result.get(0);
```

Comment:
Can we disable pushdown, run the query and then compare the results with and without pushdown?